integration/src/main/java/scripts/testscripts/DatasetPermissionOperations.java

```diff
@@ -165,8 +165,7 @@ private void testDiscovererPermissionsForSnapshotDataset() throws Exception {
         ApiException.class,
         () -> userDatasetsApi.listDatasetPreviewTables(adminTestSnapshotDatasetId));
     assertThat(
-        userDatasetsApi.getApiClient().getStatusCode(),
-        is(HttpStatusCodes.STATUS_CODE_UNAUTHORIZED));
+        userDatasetsApi.getApiClient().getStatusCode(), is(HttpStatusCodes.STATUS_CODE_FORBIDDEN));
```

Comment:
Should this test explicitly require 403, or should our test use `is4xxClientError()` instead?

Is the 403 response part of our API that we must preserve (and have full control over), or is it a way for us to expose the behavior of another service to the caller?

If it's part of another service's behavior (which I think it is), then I think our test shouldn't rely on a specific 4xx error code.

We may want to test that 4xx codes are properly propagated by our APIs, but that can be done in a unit test where both the input and output codes are controlled by the test.